cookbooks/aws-parallelcluster-config/templates/default/init/fetch_and_run.erb

```diff
@@ -21,23 +21,39 @@ function error_exit () {
   script=`basename $0`
   echo "parallelcluster: ${script} - $1"
   logger -t parallelcluster "${script} - $1"
-  echo "$1. If --rollback-on-failure was set to false, check /var/log/cfn-init.log and /var/log/cloud-init-output.log for more details." > /var/log/parallelcluster/bootstrap_error_msg
+  case $custom_script in
+    OnNodeStart|OnNodeConfigured)
+      echo "$1 Please check /var/log/cfn-init.log in the head node, or check the cfn-init.log in CloudWatch logs. Please refer to https://docs.aws.amazon.com/parallelcluster/latest/ug/troubleshooting-v3.html#troubleshooting-v3-get-logs for more details on ParallelCluster logs." > /var/log/parallelcluster/bootstrap_error_msg
+      ;;
+  esac
   exit 1
 }
 
+function process_url (){
```

Comment:
Could you add a comment in the code to explain what this function is doing and why we trim the url? What about rename it to trim_url?

Reply:
Sure I'll add a comment.
For the function name, I would prefer to still name it as "process_url" because if the url is not long, we will not trim it. This function will first check the length of the url and trim it if it's too long, and then return either the trimmed url or the original url. I consider the whole procedure as "processing the url" instead of "trimming the url". Does it make sense?

Reply:
comment added